lib/ansible/modules/cloud/amazon/aws_secret.py

```diff
@@ -0,0 +1,358 @@
+    return True
+
+
+def main():
+    module = AnsibleAWSModule(
+        argument_spec={
+            'name': dict(required=True),
+            'state': dict(choices=['present', 'absent'], default='present'),
+            'description': dict(),
+            'kms_key_id': dict(),
+            'secret_type': dict(choices=['binary', 'string'], default="string"),
+            'secret': dict(default=""),
+            'tags': dict(type='dict', default={}),
+            'rotation_lambda': dict(),
+            'rotation_interval': dict(type='int', default=30),
+            'recovery_window': dict(type='int', default=30),
+        },
+        supports_check_mode=True,
+    )
+
+    changed = False
+    state = module.params.get('state')
+    secrets_mgr = SecretsManagerInterface(module)
+    recovery_window = module.params.get('recovery_window')
+    secret = Secret(
+        module.params.get('name'),
+        module.params.get('secret_type'),
+        module.params.get('secret'),
+        description=module.params.get('description'),
+        kms_key_id=module.params.get('kms_key_id'),
+        tags=module.params.get('tags'),
+        lambda_arn=module.params.get('rotation_lambda'),
+        rotation_interval=module.params.get('rotation_interval')
+    )
+
+    current_secret = secrets_mgr.get_secret(secret.name)
+
+    if state == 'absent':
+        if current_secret:
+            if not current_secret.get("DeletedDate"):
+                result = secrets_mgr.delete_secret(secret.name, recovery_window=recovery_window)
+                changed = True
+            elif current_secret.get("DeletedDate") and recovery_window == 0:
+                result = secrets_mgr.delete_secret(secret.name, recovery_window=recovery_window)
+                changed = True
+        else:
+            result = "secret does not exist"
+
+    if state == 'present':
+        if current_secret is None:
+            result = secrets_mgr.create_secret(secret)
+            changed = True
+        else:
+            if current_secret.get("DeletedDate"):
+                secrets_mgr.restore_secret(secret.name)
+                changed = True
+            if not secrets_mgr.secrets_match(secret, current_secret):
+                result = secrets_mgr.update_secret(secret)
+                changed = True
+            if not rotation_match(secret, current_secret):
+                result = secrets_mgr.update_rotation(secret)
+                changed = True
+            current_tags = boto3_tag_list_to_ansible_dict(current_secret.get('Tags', []))
+            tags_to_add, tags_to_remove = compare_aws_tags(current_tags, secret.tags)
+            if tags_to_add:
+                secrets_mgr.tag_secret(secret.name, ansible_dict_to_boto3_tag_list(tags_to_add))
+                changed = True
+            if tags_to_remove:
+                secrets_mgr.untag_secret(secret.name, tags_to_remove)
+                changed = True
+        result = secrets_mgr.get_secret(secret.name)
```

Comment:
Use camel_dict_to_snake_dict on result for consistency with other AWS modules

Reply:
ok done